### PR TITLE
Add input field for VATIN

### PR DESCRIPTION
**Pull request by lesser1**
- Add input field to AddressInformationView.vue
- Add two new rules to check its input

[The corresponding Gropius Issue](https://frontend.gropius.duckdns.org/components/9d12d6ac-e85a-407c-bdff-a8321fb75d3a/issues/932f038d-8d1c-4259-a3fa-d51393915884)

### Definition of Done
- [x] Requirements of the issue are met
- [x] Code has been reviewed
- [x] Code is documented

### CODE REVIEW

File: src/views/checkout/AddressInformationView.vue [hunk 1]
```diff
@@ -47,6 +58,13 @@ watch(
     () => onSelectedDeliveryAddressChanged()
 )
 
+const _VATIN = ref<string | undefined>(order.value.vatNumber)
```
Comment (nk-coding) [inline src/views/checkout/AddressInformationView.vue]: is this consistent with how you named other refs?
looks a bit strange to me

File: src/util/rules.ts [hunk 2]
```diff
@@ -118,3 +118,43 @@ export function isValidCreditCardExpirationDate(
...
     return true
 }
+
+/**
+ * Checks whether the input is undefined, an empty string, or a valid VAT identification number (VATIN).
+ * @param input - The string to be validated.
+ * @returns Returns true if the input is undefined, an empty string, or a valid VATIN. Otherwise, returns a string error message indicating the validation failure.
+ */
+export function isUndefinedOrEmptyStringOrValidVATIN(
+    input: string
```
Comment (nk-coding) [inline src/util/rules.ts]: the docs say this can be undefined, however the type says sth else

File: src/util/rules.ts [hunk 3]
```diff
@@ -118,3 +118,43 @@ export function isValidCreditCardExpirationDate(
...
+ * https://en.wikipedia.org/wiki/VAT_identification_number
+ * @param input - The string to be validated as a VATIN.
+ * @returns Returns true if the input is a valid VATIN, otherwise returns a string error message indicating the validation failure.
+ */
+export function isValidVATIN(input: string): boolean | string {
+    if (typeof input !== 'string' || input.length === 0) {
+        return false
+    }
+
+    const patternToMatch = /^(EU|[A-Z]{2}|EL|XI)[\dA-Z]{2,13}$/
```
Comment (nk-coding) [inline src/util/rules.ts]: where does this come from? at least the Wikipedia article also mentions that dashes are used quite commonly, and I do not think this regex allows it

File: src/views/checkout/AddressInformationView.vue [hunk 4]
```diff
@@ -9,6 +9,16 @@
                 v-model:address="selectedDeliveryAddress"
                 @new-address-saved="triggerForQueryingOfBillingAddresses++"
             />
+            <v-text-field
+                clearable
+                label="VAT Identification Number"
+                prepend-icon="mdi-card-account-details"
+                required
```
Comment (nk-coding) [inline src/views/checkout/AddressInformationView.vue]: why is this required? the rule also seems to allow an empty string

File: src/views/checkout/AddressInformationView.vue [hunk 5]
```diff
@@ -47,6 +58,13 @@ watch(
     () => onSelectedDeliveryAddressChanged()
 )
 
+const _VATIN = ref<string | undefined>(order.value.vatNumber)
+
+watch(
+    () => _VATIN.value,
+    () => updateVATINOfOrderInformation()
```
Comment (nk-coding) [inline src/views/checkout/AddressInformationView.vue]: so, why not directly use order.value.vatNumber for the model of the input? or a computed ref at least?

**Review by nk-coding — CHANGES_REQUESTED** [review]

File: src/views/checkout/AddressInformationView.vue (hunk 1 above)
Comment (lesser1) [reply src/views/checkout/AddressInformationView.vue]: No, it's not consistent. I considered `vatIdentificationNumber`, `vatNumber`, `VATIN` and `_VATIN`. I ended up with the last option because I wanted the ref to reflect the real world's acronym which is _VATIN_ but I did not want the ref to start with an uppercase letter. At the end of the day, it's only a ref. Consistency is a beautiful thing, so I will rename it to `vatNumber` -- also, to keep it simple.

File: src/views/checkout/AddressInformationView.vue (hunk 5 above)
Comment (lesser1) [reply src/views/checkout/AddressInformationView.vue]: Since I do not have a convincing reason to keep the ref in the middle, I will change it. This then also resolves the issue regarding the naming of the `_VATIN` ref.

File: src/views/checkout/AddressInformationView.vue (hunk 4 above)
Comment (lesser1) [reply src/views/checkout/AddressInformationView.vue]: Copy-paste mistake... my bad. Will remove it.

File: src/util/rules.ts (hunk 3 above)
Comment (lesser1) [reply src/util/rules.ts]: Which article are you referring to? I could only find that Poland allows hyphens (dashes) for readability but formally their numbers only consist of digits. Of course, we could allow hyphens (dashes) and simply remove them as soon as we pass the value to the service. How about that?

File: src/util/rules.ts (hunk 2 above)
Comment (lesser1) [reply src/util/rules.ts]: Good point...

File: src/views/checkout/AddressInformationView.vue (hunk 1 above)
Comment (lesser1) [reply src/views/checkout/AddressInformationView.vue]: 00fb73f28089e2bd0ced3fa889dac139e14d6745

File: src/util/rules.ts (hunk 2 above)
Comment (lesser1) [reply src/util/rules.ts]: 00fb73f28089e2bd0ced3fa889dac139e14d6745

File: src/util/rules.ts (hunk 3 above)
Comment (lesser1) [reply src/util/rules.ts]: 00fb73f28089e2bd0ced3fa889dac139e14d6745

File: src/views/checkout/AddressInformationView.vue (hunk 4 above)
Comment (lesser1) [reply src/views/checkout/AddressInformationView.vue]: 00fb73f28089e2bd0ced3fa889dac139e14d6745

File: src/views/checkout/AddressInformationView.vue (hunk 5 above)
Comment (lesser1) [reply src/views/checkout/AddressInformationView.vue]: 00fb73f28089e2bd0ced3fa889dac139e14d6745

File: src/util/rules.ts (hunk 3 above)
Comment (nk-coding) [reply src/util/rules.ts]: the article you reference in your docs: https://en.wikipedia.org/wiki/VAT_identification_number

below, there are some non-EU countries, some use hyphen
but I don't know if these are required

**Review by nk-coding — APPROVED** [review]
